packages/journey-manager/src/index.ts

```diff
@@ -549,14 +610,20 @@ export const run = async (props: RunProps): Promise<RunOutput> => {
         }
       });
     });
-    debouncedSetHighlights();
-    // If the document changed for any reason (click, popstate event etc.), check if the URL also changed
-    // If it did, handle page load events
-    const newUrl = window.location.toString();
-    if (newUrl !== previousUrl) {
-      handleLoadSteps();
+    if (props.ui?.highlights ?? false) {
+      setHighlights();
     }
-    previousUrl = newUrl;
+    observeIntersections();
+    // This timeout here seems to solve some timing issues on URL handling
+    setTimeout(() => {
```

Comment:
Hm...

I'd like to understand a lot more about this. In my experience sprinkling `setTimeout` throughout your codebase is just a way to sweep bugs under the carpet.

What exact issues are you observing that this mitigates?

Reply:
If you take out the `setTimeout`, page load events for URL changes are completely messed up. For example: if I'm on `#tab1` and change the URL to `#tab2`, any page load event associated with `#tab2` will fire only when you leave `#tab2` to say `#tab3`. With the timeout, everything works reliably.

Reply:
I have an inkling it has to do with the timing/race condition of DOM mutations and URL changes, so with the timeout we are making sure that everything 'has settled' around the document change.